### ISSUE TITLE
`times` and `start|end|steps` simulations do not work together

### ISSUE THREAD

**Issue by matthiaskoenig** [sys-bio/roadrunner#889]
Hi all,
multiple issues/bugs when running the new `times` timevector simulations.
It does not work to run simulation with `times` and `start|end|steps` with the same model.

See minimal example below:
```python
"""Simulate roadrunner with a times vector."""

from rich import print
import roadrunner

from sbmlutils.test import REPRESSILATOR_SBML

r: roadrunner.RoadRunner = roadrunner.RoadRunner(str(REPRESSILATOR_SBML))
s1 = r.simulate(times=[0, 10.98, 50.12])
s2 = r.simulate(start=0, end=10, steps=50)

print(s1)
print("-" * 80)
print(s2)
print("-" * 80)
```

Results in 
```bash
     time,    [PX],    [PY],    [PZ],      [X],     [Y],      [Z]
 [[     0,       0,       0,       0,        0,      20,        0],
  [ 10.98, 628.103, 509.217, 137.166,  7.20079, 1.54637, 0.790015],
  [ 50.12, 313.002, 89.5364, 841.752, 0.407179, 1.12783,  14.4403]]

--------------------------------------------------------------------------------
     time,    [PX],    [PY],    [PZ],      [X],     [Y],      [Z]
 [[     0, 313.002, 89.5364, 841.752, 0.407179, 1.12783,  14.4403],
  [ 10.98, 160.609, 169.182, 993.009,  0.22754,  3.8794,  7.26332],
  [ 50.12, 296.405, 1027.34, 112.268,  7.51746, 3.30853, 0.194416]]
```

```python
r: roadrunner.RoadRunner = roadrunner.RoadRunner(str(REPRESSILATOR_SBML))
s1 = r.simulate(start=0, end=10, steps=50)
s2 = r.simulate(times=[0, 10.98, 50.12])

print(s1)
print("-" * 80)
print(s2)
print("-" * 80)
```
Results in
```
--------------------------------------------------------------------------------
    time,    [PX],    [PY],    [PZ],     [X],     [Y],      [Z]
 [[    0,       0,       0,       0,       0,      20,        0],
  [  0.2, 4.04301, 30.6426, 3.74038, 5.78591, 24.4453,  4.96978],
  [  0.4, 15.6205, 66.6337, 12.2522, 10.9648, 28.2786,  7.07722],
  [  0.6, 33.4972, 106.487,  22.331,  15.107, 30.6326,  7.66294],
  [  0.8, 55.9882, 147.767, 32.6275, 18.0331, 31.2075,  7.68678],
  [    1, 81.4404, 188.382, 42.6414, 19.9034, 30.6155,  7.49101],
  [  1.2, 108.543, 227.179, 52.1753,  20.985, 29.4545,  7.20299],
  [  1.4, 136.355,  263.65, 61.1509, 21.5114, 28.0508,  6.87656],
  [  1.6, 164.228, 297.616, 69.5427, 21.6547, 26.5631,  6.53724],
  [  1.8, 191.722, 329.056, 77.3508, 21.5344, 25.0694,   6.1981],
  [    2, 218.539, 358.027, 84.5889, 21.2335, 23.6089,  5.86625],
  [  2.2, 244.482, 384.623, 91.2779, 20.8096, 22.2017,  5.54559],
  [  2.4, 269.423, 408.959, 97.4424, 20.3033, 20.8578,  5.23823],
  [  2.6, 293.285, 431.156, 103.109, 19.7439, 19.5815,  4.94524],
  [  2.8, 316.024, 451.337, 108.304, 19.1523, 18.3741,  4.66702],
  [    3, 337.622, 469.626, 113.054, 18.5441, 17.2349,  4.40362],
  [  3.2, 358.081,  486.14, 117.387, 17.9305,  16.162,  4.15478],
  [  3.4, 377.415, 500.995, 121.328, 17.3199, 15.1532,   3.9201],
  [  3.6, 395.647, 514.299,   124.9, 16.7185, 14.2055,  3.69909],
  [  3.8,  412.81, 526.155, 128.128, 16.1307, 13.3162,  3.49119],
  [    4, 428.938, 536.663, 131.035, 15.5598,  12.482,   3.2958],
  [  4.2, 444.072, 545.913,  133.64, 15.0081, 11.7001,  3.11232],
  [  4.4, 458.251, 553.995, 135.965, 14.4773, 10.9675,  2.94015],
  [  4.6,  471.52, 560.988, 138.028, 13.9683, 10.2814,  2.77871],
  [  4.8, 483.921, 566.972, 139.848, 13.4818,   9.639,  2.62739],
  [    5, 495.497, 572.017, 141.441, 13.0179, 9.03766,  2.48566],
  [  5.2,  506.29, 576.192, 142.823, 12.5768, 8.47492,  2.35297],
  [  5.4, 516.344, 579.559,  144.01, 12.1582, 7.94839,   2.2288],
  [  5.6, 525.698,  582.18, 145.015, 11.7618, 7.45582,  2.11266],
  [  5.8, 534.392, 584.108, 145.851, 11.3871, 6.99511,   2.0041],
  [    6, 542.466, 585.396, 146.531, 11.0336, 6.56424,  1.90265],
  [  6.2, 549.956, 586.093, 147.067, 10.7007, 6.16133,  1.80791],
  [  6.4, 556.898, 586.244,  147.47, 10.3877,  5.7846,  1.71947],
  [  6.6, 563.327, 585.891, 147.749, 10.0939,  5.4324,  1.63697],
  [  6.8, 569.277, 585.074, 147.915, 9.81877, 5.10315,  1.56004],
  [    7, 574.777,  583.83, 147.977, 9.56147, 4.79537,  1.48836],
  [  7.2,  579.86, 582.193, 147.942, 9.32134,  4.5077,  1.42162],
  [  7.4, 584.554, 580.196, 147.819, 9.09769, 4.23883,  1.35951],
  [  7.6, 588.886, 577.869, 147.615, 8.88984, 3.98755,  1.30177],
  [  7.8, 592.883, 575.239, 147.338, 8.69711, 3.75273,  1.24813],
  [    8,  596.57, 572.333, 146.993, 8.51886, 3.53329,  1.19835],
  [  8.2,  599.97, 569.175, 146.587, 8.35445, 3.32824,   1.1522],
  [  8.4, 603.106, 565.787, 146.125, 8.20327, 3.13665,  1.10947],
  [  8.6, 605.999, 562.191, 145.613, 8.06471, 2.95763,  1.06995],
  [  8.8, 608.671, 558.407, 145.056, 7.93822, 2.79037,  1.03346],
  [    9, 611.139, 554.451, 144.458, 7.82323, 2.63411, 0.999826],
  [  9.2, 613.423, 550.343, 143.824,  7.7192, 2.48811, 0.968876],
  [  9.4, 615.539, 546.096, 143.158, 7.62564, 2.35172, 0.940458],
  [  9.6, 617.504, 541.727, 142.464, 7.54204,  2.2243, 0.914428],
  [  9.8, 619.334, 537.248, 141.745, 7.46794, 2.10527, 0.890652],
  [   10, 621.043, 532.672, 141.004, 7.40287, 1.99407, 0.869004]]

--------------------------------------------------------------------------------
    time,    [PX],    [PY],    [PZ],     [X],     [Y],      [Z]
 [[    0, 621.043, 532.672, 141.004, 7.40287, 1.99407, 0.869004],
  [  0.2, 622.645, 528.012, 140.245, 7.34641, 1.89019, 0.849367],
  [  0.4, 624.153, 523.277, 139.472, 7.29813, 1.79314, 0.831626]]

--------------------------------------------------------------------------------
```

* if first running a `times` simulation the second simulation still has the times vecor as time points instead of `[0, 0.2, 0.4, ... 50]`.
* if first running a steps simulation the `times` simulation has the incorrect time points `[0, 0.2, 0.4] which should be `[0, 10.98, 50.12]`.

This is a major issue because incorrect time points and incorrect number of time points are returned from the simulator!
[times_vector_bug.zip](https://github.com/sys-bio/roadrunner/files/7359688/times_vector_bug.zip)

**Comment by luciansmith** [comment]
This is now fixed--tested with this:

```
import tellurium as te
import roadrunner

r = te.loada("""
species a=3
""")

r1 = r.simulate()
r2 = r.simulate(times=[3, 5, 58])
r3 = r.simulate(0, 10, 11)
r4 = r.simulate(times=[3, 5, 58])
r5 = r.simulate()

print(len(r1))
print(r2)
print(r3)
print(r4)
print(len(r5))
```
Thank you for reporting the problem!

**Comment by matthiaskoenig** [comment]
Thanks. This works. Just tested with the azure artifacts.
Would it be possible to release a libroadrunner-experimental with this fix?

**Comment by luciansmith** [comment]
This is now released in 2.1.3, for both libroadrunner and libroadrunner-experimental.  (Since it was a bug fix, I put it in the main release.)
